### ISSUE TITLE
False positive for array access to mixed in conditional with `||`

### ISSUE THREAD

**Issue by TysonAndre** [vimeo/psalm#375]
Expected: No PossiblyUndefinedVariable. Maybe a warning related to mixed, or no warnings.
Observed: PossiblyUndefinedVariable for accessing array field of `mixed $array`

Possibly related to #374 (which may also have inferred type of `mixed`

```php
<?php
/** @param mixed $array (bug doesn't happen for array $array) */
function check_array($array) : bool {
    if (is_null($array['field']) || trim($array['field']) === '') {  // Unexpected PossiblyUndefinedVariable - Found a possibly undefined variable when evaluating $array['field']

        return false;
    }

    return true;
}
```

**Comment by weirdan** [comment]
`$array['field']` _is_ possibly undefined in this case. PossiblyUndefined*Variable* is confusing though, as it's possibly undefined _array index_ here.

**Comment by weirdan** [comment]
```
$ cat qq.php
<?php

function check_array($array) : bool {
    if (is_null($array['field']) || trim($array['field']) === '') {

        return false;
    }

    return true;
}

check_array(["blahblahblah" => false]);
$ php qq.php
PHP Notice:  Undefined index: field in /home/weirdan/src/t/qq.php on line 4
PHP Stack trace:
PHP   1. {main}() /home/weirdan/src/t/qq.php:0
PHP   2. check_array() /home/weirdan/src/t/qq.php:12
```

**Comment by weirdan** [comment]
Or it could be me expecting too much from Psalm this time :)

**Comment by muglug** [comment]
Fixed in https://github.com/vimeo/psalm/commit/d10898b853b6e3f264c9477cab07a9a13de01f58

Basically, I inserted that opportunistic issue, thinking it would catch some cool things. It didn't, really, and messed up loop analysis quite a bit.
